rfi_file_monitor/operations/scicataloguer.py

```diff
@@ -0,0 +1,804 @@
+                vexpand=False,
+            ),
+            0,
+            4,
+            1,
+            1,
+        )
+        
+
+        # create combo box
+        
+        combo = Gtk.ComboBoxText.new()
+        for k in self.instr_dict["techniques"].keys():
+            combo.append_text(k)
+        if len(self.instr_dict["techniques"].keys()) == 1:
+            combo.set_active(0)
+
+        widget = self.register_widget(combo, "technique")
+        self._grid.attach(widget, 1, 4, 1, 1)
+
+        self.parser_list = []
+        for e in importlib.metadata.entry_points()[
+            "rfi_file_monitor_extensions.metadataparsers"
+        ]:
+            self.parser_list.append(e.load())
+        """
+
+        # Dataset name
+        self._grid.attach(
+            Gtk.Label(
+                label="Experiment name",
+                halign=Gtk.Align.START,
+                valign=Gtk.Align.CENTER,
+                hexpand=False,
+                vexpand=False,
+            ),
+            2,
+            4,
+            1,
+            1,
+        )
+        self._exp_name_entry = self.register_widget(
+            Gtk.Entry(
+                halign=Gtk.Align.FILL,
+                valign=Gtk.Align.CENTER,
+                hexpand=True,
+                vexpand=False,
+            ),
+            "experiment_name",
+        )
+        self._grid.attach(self._exp_name_entry, 3, 4, 1, 1)
+
+        # Instrument
+        # TO DO - this is temporary until instrument preferences configured
+        self._grid.attach(
+            Gtk.Label(
+                label="Instrument",
+                halign=Gtk.Align.START,
+                valign=Gtk.Align.CENTER,
+                hexpand=False,
+                vexpand=False,
+            ),
+            0,
+            5,
+            1,
+            1,
+        )
+        self._instrument_entry = self.register_widget(
+            Gtk.Entry(
+                halign=Gtk.Align.FILL,
+                valign=Gtk.Align.CENTER,
+                hexpand=True,
+                vexpand=False,
+            ),
+            "instrument_choice",
+        )
+        self._grid.attach(self._instrument_entry, 1, 5, 1, 1)
+
+        # Technique
+        self._grid.attach(
+            Gtk.Label(
+                label="Technique",
+                halign=Gtk.Align.START,
+                valign=Gtk.Align.CENTER,
+                hexpand=False,
+                vexpand=False,
+            ),
+            2,
+            5,
+            1,
+            1,
+        )
+        self._technique_entry = self.register_widget(
+            Gtk.Entry(
+                halign=Gtk.Align.FILL,
+                valign=Gtk.Align.CENTER,
+                hexpand=True,
+                vexpand=False,
+            ),
+            "technique",
+        )
+        self._grid.attach(self._technique_entry, 3, 5, 1, 1)
+
+        # Input boxes for derived dataset specific fields
+        self._grid.attach(
```

Comment:
Wonder if we could consider making it a bit clearer in the layout that this is for derived data sets only.

Maybe we can have the derived dataset checkbox in it's own line and then its optional dependencies below it.

It would be nice if these could be greyed out when the Derived Dataset checkbox  is not clicked and then activated when it is.